machinelearning/llama.cpp.md

```diff
@@ -0,0 +1,73 @@
+# Large Language Model (LLM) inference on Graviton CPUs with llama.cpp
+
+**Introduction**
+
+The main goal of [llama.cpp](https://github.com/ggerganov/llama.cpp) is to enable LLM inference with minimal setup and state-of-the-art performance on a wide variety of hardware. It's a plain C/C++ implementation without any dependencies. It supports quantized general matrix multiply and add (GEMM) kernels for faster inference and reduced memory use. The quantized GEMM kernels are optimized for AWS Graviton processors using Arm Neon and SVE based matrix multiply and add (MMLA) instructions. This document covers how to build and run llama.cpp efficiently for LLM inference on AWS Graviton based Amazon EC2 instances.
+
+# How to use llama.cpp on Graviton CPUs
+
+Building from sources is the recommended way to use llama.cpp on Graviton CPUs, and for other hardware platforms too. This section provides the instructions on how to build it from sources.
+
+**Prerequisites**
+
+It is highly recommended to use the AMIs based on Linux Kernel 5.10 and beyond for the best LLM inference performance on Graviton3 instances. The following queries can be used to list down the AMIs with Kernel 5.10 and beyond.
```

Comment:
I don't love two sentences in a row using the passive voice. I think the first one is probably right to be written with the passive voice the emphasis is that "it is highly recommended". The second one you could just say this:

```suggestion
It is highly recommended to use the AMIs based on Linux Kernel 5.10 and beyond for the best LLM inference performance on Graviton3 instances. Use the following queries to list down the AMIs with Kernel 5.10 and beyond.
```

But frankly, it doesn't matter that much.

Reply:
you should say why....